### PR TITLE
option to pass I2C object

### PR DESCRIPTION
**Pull request by pilotak**
Ability to pass I2C object

### CODE REVIEW

File: I2CEEBlockDevice.cpp [hunk 1]
```diff
@@ -21,11 +21,26 @@
 I2CEEBlockDevice::I2CEEBlockDevice(
         PinName sda, PinName scl, uint8_t addr,
         bd_size_t size, bd_size_t block, int freq)
-    : _i2c(sda, scl), _i2c_addr(addr), _size(size), _block(block)
+    : _i2c_p(new I2C(sda, scl)), _i2c(*_i2c_p), _i2c_addr(addr),
```
Comment (geky) [inline I2CEEBlockDevice.cpp]: Could you instead use [placement new](https://en.wikipedia.org/wiki/Placement_syntax)? That way the class still uses the memory needed for the I2C object and we don't have to rely on dynamic memory:

``` c++
uint32_t _i2c_buffer[sizeof(I2C) / sizeof(uint32_t);
I2C *_i2c;
_i2c = new (_i2c_buffer) I2C(sda, scl);
_i2c->~I2C(); // note needs manual delete
```

**Comment by pilotak** [comment]
I don't understand the `_i2c_buffer` bit, but it works without it fine

**Comment by pilotak** [comment]
can this be merged?

**Comment by geky** [comment]
@pilotak Oh sorry! This needs to be using placement new so static memory errors get caught at compile time. Mind if I modify your pr?

File: I2CEEBlockDevice.h [hunk 2]
```diff
@@ -70,7 +70,24 @@ class I2CEEBlockDevice : public BlockDevice {
...
+     /** Constructor to create an I2CEEBlockDevice on I2C pins
+     *
+     *  @param i2c      The I2C instance
+     *  @param addr     The 8bit I2C address of the chip, common range 0xa0 - 0xae.
+     *  @param size     The size of the device in bytes
+     *  @param block    The page size of the device in bytes, defaults to 32bytes
+     *  @param freq     The frequency of the I2C bus, defaults to 400K.
+     */
+    I2CEEBlockDevice(
+            I2C &i2c_obj, uint8_t address,
```
Comment (geky) [inline I2CEEBlockDevice.h]: nit: Using a pointer here would make it clear the I2C object is being borrowed.

**Comment by geky** [comment]
Oh you're fast. I added the placement new change.

@pilotak does this look good to you?

**Comment by pilotak** [comment]
that's perfect, this way you can actually see how much space it takes during compile

**Pull request merged**